### PR TITLE
[flutter_plugin_tools] Fix subpackage analysis

### PR DESCRIPTION
**Pull request by stuartmorgan**
Fixes a regression in `analyze` when packages have non-example
subpackages. Restructuring in 0.8.0 accidentally caused them not to have
their dependencies fetched, causing analysis failures.

Unblocks rolling flutter/packages to the latest tooling.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [ ] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by stuartmorgan** [comment]
I was thinking analysis was a run-on-packages-and-subpackages iteration when I made the previous change; I forget that it's actually iterating over just the top-level packages, but that `analyze` *automatically* runs over subpackages within the directory.

**Comment by ditman** [comment]
```
./script/tool_runner.sh make-deps-path-based --target-dependencies-with-non-breaking-updates
Finding changed packages relative to "6852d0a311a8c501b1718ca54b382b97c4db1a22"...
Rewriting references to: flutter_plugin_tools...
�[31mUnable to find package "flutter_plugin_tools"�[0m

Exit status: 3
```

**Comment by stuartmorgan** [comment]
Doh

**Comment by stuartmorgan** [comment]
Should be fixed now; I added logic to skip packages that aren't in the packages directory when finding changed packages to rewrite.

**Comment by ditman** [comment]
Ah, good point, I didn't remember that `tools` was outside of `packages` (x_x)

File: script/tool/lib/src/analyze_command.dart [hunk 1]
```diff
@@ -102,16 +102,25 @@ class AnalyzeCommand extends PackageLoopingCommand {
...
+      package,
+      ...await getSubpackages(package).toList(),
+    ];
+    for (final RepositoryPackage packageToGet in packagesToGet) {
+      if (packageToGet.directory.basename != 'example' ||
+          !RepositoryPackage(packageToGet.directory.parent)
+              .pubspecFile
+              .existsSync()) {
+        final int exitCode = await processRunner.runAndStream(
+            flutterCommand, <String>['packages', 'get'],
```
Comment (ditman) [inline script/tool/lib/src/analyze_command.dart:119]: This is extremely nitpicky, I'm sure I've not noticed it in dozens of other PRs!, but `flutter packages` is an undocumented alias (kept for backwards compatibility, IIRC) of: `flutter pub` 🤓

```
$ flutter --help | grep packages
  pub               Commands for managing Flutter packages.
```

(It doesn't show up even in `-v` mode of help!, but there's a `--packages` that has a different purpose.)

Comment (ditman) [reply script/tool/lib/src/analyze_command.dart:119]: (`flutter packages get` will probably work forever, but I'm flagging it just in case 😛)

**Review by ditman — APPROVED** [review]
🚀 LGTM! 🛸

**Pull request merged**